### ISSUE TITLE
Funcotator datasource configs should not honor spaces (and some other special characters) in the datasource name nor version.

### ISSUE THREAD

**Issue by LeeTL1220** [broadinstitute/gatk#5937]
### Affected tool(s) or class(es)
Funcotator

### Affected version(s)
All

### Description 
Since we expect users to write config files manually, we need to have some enforcement of naming rules.  At the least, to disallow spaces in the name and version fields.  There are many places throughout the code where we assume that there will be no spaces.  Additionally, I hear from users that they want any Funcotator tsv outputs to never have spaces (or tabs or other special characters -- "_", "-" are obviously okay).  

We can solicit users about which special characters are okay, but definitely disallow spaces and tabs.

#### Steps to reproduce
Add a space to the Gencode datasource config (name or version field) and try to funcotate a segment file.

#### Expected behavior
No errors and no spaces in the field names.

#### Actual behavior
Exception in gene list output renderer.

**Comment by jonn-smith** [comment]
@LeeTL1220 It sounds like you're suggesting two things.  

First, if a datasource config has a space in it, throw an error and tell the user to remove the space.  

Second - remove spaces from funcotation field names when outputting the funcotations (could still happen with XSVs with spaces in column headers).

Is this correct?

**Comment by LeeTL1220** [comment]
> First, if a datasource config has a space in it, throw an error and tell the user to remove the space.

Yes, but trim trailing and leading spaces first.  Do this when initializing the datasource.  So if a config file specifies a leading or trailing whitespace, it is ignored.

> Second - remove spaces from funcotation field names when outputting the funcotations (could still happen with XSVs with spaces in column headers).

Actually, since we know the output field names before we do any work, I would recommend doing something similar as above (i.e. Remove trailing or leading spaces and throw error if any other spaces are found).
